src/components/common/UserProfileEdit/UserAdvancedSettings.tsx

```diff
@@ -109,134 +106,53 @@ const UserAdvancedSettings = () => {
-                }}
-                name={SlotKey.Metatransactions}
-                disabled={!metatransactionsToggleAvailable}
-              />
-              <QuestionMarkTooltip
-                tooltipText={MSG.metaTooltip}
-                tooltipTextValues={{ br: <br /> }}
-                className={stylesAdvance.tooltipContainer}
-                tooltipClassName={stylesAdvance.tooltipContent}
-                tooltipPopperOptions={{
-                  placement: 'right',
-                }}
-              />
-            </div>
-            <p className={stylesAdvance.descriptions}>
-              <FormattedMessage {...MSG.metaDescription} />
-            </p>
-            {!metatransactionsToggleAvailable && (
-              <div className={stylesAdvance.metaDesc}>
-                <FormattedMessage {...MSG.metaDescGlobalOff} />
-              </div>
-            )}
-            <hr />
-            <div className={stylesAdvance.sectionTitle}>
-              <FormattedMessage
-                {...MSG.customEndpoints}
-                values={{ isOn: values[SlotKey.DecentralizedMode] }}
-              />
-            </div>
-            <p className={stylesAdvance.descriptions}>
-              <FormattedMessage {...MSG.metaDescription} />
-            </p>
-            <div className={stylesAdvance.toggleContainer}>
-              <Toggle
-                label={MSG.labelRPC}
-                labelValues={{ isOn: decentralizedModeEnabled }}
-                name={SlotKey.DecentralizedMode}
-              />
-              <QuestionMarkTooltip
-                tooltipText={MSG.RPCTooltip}
-                className={stylesAdvance.tooltipContainer}
-                tooltipClassName={stylesAdvance.tooltipContent}
-                tooltipPopperOptions={{
-                  placement: 'right',
-                }}
-              />
-            </div>
-            <Input
-              name={SlotKey.CustomRPC}
-              disabled={!values[SlotKey.DecentralizedMode]}
-              formattingOptions={noSpaces}
-              value={customRpc}
-            />
-            <div className={stylesAdvance.validateButtonContainer}>
-              <Button
-                text={MSG.validate}
-                disabled={!values[SlotKey.DecentralizedMode] || !isValid}
-                // @TODO: validation to be added later
+    <>
+      <AdvancedSettingsHeading />
+      <Form<FormValues>
+        defaultValues={{
+          [SlotKey.Metatransactions]: metatransasctionsDefault,
+          [SlotKey.DecentralizedMode]: decentralizedModeEnabled,
+          [SlotKey.CustomRPC]: customRpc,
+        }}
+        validationSchema={validationSchema}
+        onSubmit={(values) => handleSubmit(values, setSettingsKey)}
+        resetOnSubmit
+      >
+        {({ formState: { isValid, defaultValues }, watch }) => {
+          const values = watch();
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Even though `defaultValues` correctly infers the values inputted to the Form prop, it still types it as `...values | undefined`. Hence need to assert non-null.

This invites a question of when to use the `defaultValues` provided by HookForm like here, and when to move the defaultValues provided as props to a variable and use those (like in `UserMainSettings`). 

I would prefer to use the one provided by HookForm, but it involves disabling the linter if you use a non-null assertion like here, which may get a bit ugly if you do it a lot. Hence my opting for the other pattern in `UserMainSettings`, since I'm using the default values in more than one place.

You could also do `defaultValues || {}` but this is unintuitive since we know the default Values are defined because we're defining them.